### PR TITLE
Feature sumo vehicle differentiation 

### PR DESCRIPTION
**Pull request by DavidMansolino**
Linked with omichel/webots#3961

### CODE REVIEW

**Comment by DavidMansolino** [comment]
:+1:

File: automobile/sumo-interface.md [hunk 1]
```diff
@@ -30,18 +32,24 @@ field `controllerArgs` in order to customize the behavior of the interface:
...
+| --d or --directory       | Specifies the directory where are located the SUMO network files                                                                     | By default the files should be located close to the world in a directory called `worldName_net`  |
+| --p --port               | Specifies which port SUMO and the interface should use to communicate                                                                | By default, the port 8873 is used                                                                |
+| --s --seed               | Specifies the seed of the SUMO random number generator (0 for the '--random' option of SUMO)                                         | By default, a seed of 1 is used                                                                  |
+| --useSumoDisplay         | Specifies if the gui view of SUMO should be displayed in a Webots display (only working in gui mode, and the Supervisor must have a display named 'sumo display')                                         | By default, the gui view is not displayed  |
```
Comment (omichel) [inline automobile/sumo-interface.md]: GUI should be written in uppercase as you did it for the description of the --nogui option.

File: automobile/sumo-interface.md [hunk 2]
```diff
@@ -30,18 +32,24 @@ field `controllerArgs` in order to customize the behavior of the interface:
...
+| --sumoDisplayRefreshRate | Specifies the refresh rate of the SUMO display (no effect if the SUMO display is not activated)                                      | By default, a refresh rate of 1000ms is used                                                                  |
```
Comment (omichel) [inline automobile/sumo-interface.md]: &rarr; Specifies the refresh rate of the SUMO display, **expressed in milliseconds**

File: automobile/sumo-interface.md [hunk 3]
```diff
@@ -30,18 +32,24 @@ field `controllerArgs` in order to customize the behavior of the interface:
...
+| --sumoDisplayZoom        | Specifies the initial zoom of the SUMO display in Webots (100 means no scaling)                                                      | By default, no scaling is performed                                                                  |
```
Comment (omichel) [inline automobile/sumo-interface.md]: I would use 1.0 for no scaling... (this is more consistent with what we are using generally in Webots).

File: automobile/sumo-interface.md [hunk 4]
```diff
@@ -30,18 +32,24 @@ field `controllerArgs` in order to customize the behavior of the interface:
...
+| --sumoDisplayResizeImage | Specifies if the image should be resized to fit the SUMO display size or not                                                         | By default, the image is not resized                                                                  |
```
Comment (omichel) [inline automobile/sumo-interface.md]: This is not clear how this interact with the display zoom described above...
Does the resize adjust the width / height aspect ratio? Or is it an automatic scaling that doesn't affect the aspect ratio?

**Comment by DavidMansolino** [comment]
:+1:

File: automobile/sumo-interface.md [hunk 5]
```diff
@@ -30,25 +35,31 @@ field `controllerArgs` in order to customize the behavior of the interface:
...
 %end
 
-In addition to these arguments, the plugin mechanism can be used to extend the
-interface. The plugin should be written in python, be in the same folder than
-the SUMO network files, and should implement the `SumoSupervisorPlugin` class
-with the two following entry-point functions:
+
+## Plugin mechanism
+
+In addition to the `sumo_supervisor` controller arguments, the plugin mechanism can be used to extend the interface. The plugin should be written in python, be in the same folder than the SUMO network files, and should implement the `SumoSupervisorPlugin` class with the two following entry-point functions:
```
Comment (omichel) [inline automobile/sumo-interface.md]: ...same folder **as** the SUMO network files, ....

File: automobile/sumo-interface.md [hunk 6]
```diff
@@ -30,25 +35,31 @@ field `controllerArgs` in order to customize the behavior of the interface:
...
+| --enable-height          | Specifies if height information should be extracted from the edge name                                                               | By default, all the vehicles are in the plane defined by y = 0                                   |
+| --d or --directory       | Specifies the directory where are located the SUMO network files                                                                     | By default the files should be located close to the world in a directory called `worldName_net`  |
+| --p --port               | Specifies which port SUMO and the interface should use to communicate                                                                | By default, the port 8873 is used                                                                |
+| --s --seed               | Specifies the seed of the SUMO random number generator (0 for the '--random' option of SUMO)                                         | By default, a seed of 1 is used                                                                  |
```
Comment (omichel) [inline automobile/sumo-interface.md]: I would remove the short versions of these options, e.g., `--d`, `--p` and `--s`.

**Comment by omichel** [comment]
:+1:

**Comment by DavidMansolino** [comment]
:+1:

**Pull request merged**